src/expr/attribute_internals.h

```diff
@@ -102,6 +103,29 @@ struct KindValueToTableValueMapping {
   inline static T convertBack(const T& t) { return t; }
 };
 
+/**
+ * This converts arbitrary unsigned integers (up to 64-bit) to and from 64-bit
+ * integers s.t. they can be stored in the hash table for integral-valued
+ * attributes.
+ */
+template <class T>
+struct KindValueToTableValueMapping<
+    T,
+    typename std::enable_if<std::is_unsigned<T>::value>::type>
```

Comment:
Add a comment to document what is going on here since it's not obvious if you read the code the first time.